### PR TITLE
Fix highlight of commit hash in git branches preview

### PR DESCRIPTION
**Pull request by petobens**
# Description

In repos with many commits, git log flag `--abbrev-commit` will increase the number of characters from 7 to 8 to ensure unicity. The git branches preview highlight is therefore off by one in this case.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Before and after:
image
image

**Configuration**:
* Neovim version (nvim --version): NVIM v0.10.0-dev-2361+ga376d979bd
* Operating system and version: Arch Linux

# Checklist:

- [x] My code follows the style guidelines of this project (stylua)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (lua annotations)

### CODE REVIEW

**Comment by jamestrew** [comment]
Thanks, you're right but by extension, the bigger the repo, the larger the commit hash abbreviations will get.
image

Are you able to come up with a more dynamic approach?

**Comment by petobens** [comment]
Hi @jamestrew. I added a new regex to dynamically obtain the end char. Hope it helps

File: lua/telescope/previewers/buffer_previewer.lua [hunk 1]
```diff
@@ -748,8 +748,8 @@ previewers.git_branch_log = defaulter(function(opts)
     for i = 1, #content do
       local line = content[i]
       local _, hstart = line:find "[%*%s|]*"
+      local _, hend, _ = line:find "[%*%s|]*%s(%w*)"
```
Comment (jamestrew) [inline lua/telescope/previewers/buffer_previewer.lua:751]:
```suggestion
      local hstart, hend = line:find "[0-9a-fA-F]+"
```
Do you think this simplification will work? I believe that the commit hash is always the first hex string.

Comment (petobens) [reply lua/telescope/previewers/buffer_previewer.lua]: Yep that's way better.

**Review by jamestrew — APPROVED** [review]
thanks!

**Pull request merged**